pkg/bucketclass/validation.go

```diff
@@ -1,23 +1,77 @@
 package bucketclass
 
 import (
+	"fmt"
+
 	nbv1 "github.com/noobaa/noobaa-operator/v5/pkg/apis/noobaa/v1alpha1"
 	"github.com/noobaa/noobaa-operator/v5/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // ValidateBucketClass bucket class
 func ValidateBucketClass(bc *nbv1.BucketClass) error {
 	if bc == nil {
 		return nil
 	}
-	return validateQuotaConfig(bc.Name, bc.Spec.Quota)
+	return ValidateQuotaConfig(bc.Name, bc.Spec.Quota)
 }
 
-// Validate quota config
-func validateQuotaConfig(bcName string, quota *nbv1.Quota) error {
+// ValidateQuotaConfig validates the quota values
+func ValidateQuotaConfig(bcName string, quota *nbv1.Quota) error {
 	if quota == nil {
 		return nil
 	}
 
 	return util.ValidateQuotaConfig(bcName, quota.MaxSize, quota.MaxObjects)
 }
+
+// ValidateTiersNumber validates the provided number of tiers is 1 or 2
+func ValidateTiersNumber(tiers []nbv1.Tier) error {
+	if len(tiers) != 1 && len(tiers) != 2 {
+		return util.ValidationError{
+			Msg: "unsupported number of tiers, bucketclass supports only 1 or 2 tiers",
+		}
+	}
+	return nil
+}
+
+// GetBucketclassNamespaceStoreArray returns an array of namespacestores of the provided bc
+func GetBucketclassNamespaceStoreArray(bc *nbv1.BucketClass) []string {
+	var namespaceStoresArr []string
+
+	switch bc.Spec.NamespacePolicy.Type {
+	case nbv1.NSBucketClassTypeCache:
+		namespaceStoresArr = append(namespaceStoresArr, bc.Spec.NamespacePolicy.Cache.HubResource)
+	case nbv1.NSBucketClassTypeMulti:
+		namespaceStoresArr = append(bc.Spec.NamespacePolicy.Multi.ReadResources,
+			bc.Spec.NamespacePolicy.Multi.WriteResource)
+	case nbv1.NSBucketClassTypeSingle:
+		namespaceStoresArr = append(namespaceStoresArr, bc.Spec.NamespacePolicy.Single.Resource)
+	}
+
+	return namespaceStoresArr
+}
+
+// ValidateNSFSSingleBC validates that bucketclass configured to NS of type NSFS it will only be of type Single.
+func ValidateNSFSSingleBC(bc *nbv1.BucketClass) error {
+	namespaceStoresArr := GetBucketclassNamespaceStoreArray(bc)
+
+	for _, name := range namespaceStoresArr {
+		nsStore := &nbv1.NamespaceStore{
+			TypeMeta: metav1.TypeMeta{Kind: "NamespaceStore"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: bc.Namespace,
+			},
+		}
+		if !util.KubeCheck(nsStore) {
+			return fmt.Errorf("failed to check namespacestore in NSFS single bucketclass type validation")
+		}
+		if nsStore.Spec.Type == nbv1.NSStoreTypeNSFS && bc.Spec.NamespacePolicy.Type != nbv1.NSBucketClassTypeSingle {
```

Comment:
move `bc.Spec.NamespacePolicy.Type != nbv1.NSBucketClassTypeSingle`  before the first line of this function so you won't even collect the namespacestores and do this test if this bucketclass is of type single